pkgs/os-specific/linux/kernel/generic.nix

```diff
@@ -63,9 +63,11 @@ lib.makeOverridable ({ # The kernel source tarball.
 , ignoreConfigErrors ? stdenv.hostPlatform.linux-kernel.name != "pc"
 , extraMeta ? {}
 
-, isZen      ? false
-, isLibre    ? false
-, isHardened ? false
+, isZen       ? false
+, isLibre     ? false
+, isHardened  ? false
+, isCachy     ? false
+, isCachyMain ? false
```

Comment:
No point for this, since you're using `structuredExtraConfig`, the CONFIG_CACHY will be exposed in `passthru.config`. You can then, in modules or your config, check for it in a more flexible way:

```nix
(config.boot.kernelPackages.kernel.passthru.config.CONFIG_CACHY or "n") == "y"
```

Reply:
`isCachy` is redundant, so that can be removed.

`isCachyMain` is needed to prevent some out-of-tree modules being built that are included in the CachyOS patches.

Reply:
> isCachyMain is needed to prevent some out-of-tree modules being built that are included in the CachyOS patches.

You could just check the kernel version, if it has CONFIG_CACHY and is 6.6 it would be the current LTS.

Reply:
That could work, but then there's potentially multiple places that the version has to be changed if/when the LTS kernel is bumped to a new minor release. The current way doesn't require any additional work.

Reply:
However, looking at https://github.com/NixOS/nixpkgs/blob/e4f73cb480338f34daded86d31f72464eb71c552/pkgs/os-specific/linux/sysdig/default.nix#L192-L193 it seems there's some precedent for using the `isSomeKernel` variables to mark packages as broken for at least one package.

Reply:
Yes. Kernel modules won't be broken because they're building against ""cachyos-lts"". It will be either because of the Cachy's patches, or because of the ""majorMinor"" being unsupported.

Reply:
Honestly, `isCachy` is more readable to me than that config check and it's in agreement in convention, even if technically redundant.